src/workerd/io/worker.c++

```diff
@@ -2303,6 +2305,39 @@ public:
     }
   }
 
+  kj::Promise<void> dispatchProtocolMessages(kj::MutexGuarded<MessageQueue>& incomingQueue) {
+    auto i = kj::atomicAddRef(*this->state.lockExclusive()->get()->isolate);
+    auto asyncLock = co_await i->takeAsyncLockWithoutRequest(nullptr);
+
+    auto lockedState = this->state.lockExclusive();
+    v8_inspector::V8InspectorSession& session = *lockedState->get()->session;
+    Isolate& isolate = const_cast<Isolate&>(*lockedState->get()->isolate);
+    jsg::V8StackScope stackScope;
+    Isolate::Impl::Lock recordedLock(isolate, asyncLock, stackScope);
+    for (;;) {
+      kj::String message;
+      {
+        auto lockedMessageQueue = incomingQueue.lockExclusive();
+        if (lockedMessageQueue->status != MessageQueue::Status::ACTIVE) {
+          co_return;
+        }
+
+        if (lockedMessageQueue->head == lockedMessageQueue->messages.size()) {
+          co_return;
+        }
+
+        message = kj::mv(lockedMessageQueue->messages[lockedMessageQueue->head++]);
```

Comment:
Assuming it's possible that there are multiple messages in the queue at this point, would it make sense to grab and dispatch as many as possible here in a batch while we have the lock rather than grabbing one at a time?